pkg/printers/testing/common.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package testing
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/cli-utils/pkg/apply/event"
+	"sigs.k8s.io/cli-utils/pkg/common"
+	"sigs.k8s.io/cli-utils/pkg/object"
+	printcommon "sigs.k8s.io/cli-utils/pkg/print/common"
+	"sigs.k8s.io/cli-utils/pkg/print/stats"
+	"sigs.k8s.io/cli-utils/pkg/printers/printer"
+)
+
+type PrinterFactoryFunc func() printer.Printer
+
+func PrintResultErrorTest(t *testing.T, f PrinterFactoryFunc) {
```

Comment:
I see PrintResultErrorTest is used by the list and table, but I don't see one for the json printer.

Reply:
It runs the tests for the `table` and the base printer. Both the `events` and the `json` printer uses the base printer (just uses different formatters), so they are both being tested.

Reply:
gotcha